File: 연습문제/추억점수/추억점수.java

import java.util.HashMap;
import java.util.Map;
// key : 이름(String), value : 점수(Integer)
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
      // 이름 배열 name과 점수 배열 yearning의 인덱스를 순회하며 HashMap에 저장
        Map<String, Integer> point = new HashMap<>();

        // 이름과 그리움 점수 매핑
        for (int i = 0; i < name.length; i++) {
            point.put(name[i], yearning[i]);
        }

        // 사진별 점수 계산
        for (int i = 0; i < photo.length; i++) {
            int score = 0; // 점수 초기화  
            for (String person : photo[i]) {
                score += point.getOrDefault(person, 0); // 없는 사람은 기본값 0
            }
            answer[i] = score; // 사진별 점수 저장
        }

        return answer;
    }
}

File: 연습문제/추억점수js/추억점수.js
function solution(name, yearning, photo) {
    // Step 1: 이름과 점수 매핑
    const point = {};
    for (let i = 0; i < name.length; i++) {
        point[name[i]] = yearning[i];
    }

    // Step 2: 사진별 점수 계산
    const answer = photo.map(group => {
        return group.reduce((score, person) => {
            // 점수 합산, 없는 경우 기본값 0
            return score + (point[person] || 0);
        }, 0);
    });

    return answer;
}
